codewof/codewof/codewof_utils.py

```diff
@@ -0,0 +1,207 @@
+import datetime
+import json
+import logging
+from dateutil.relativedelta import relativedelta
+
+from programming.models import (
+    Profile,
+    Question,
+    TestCase,
+    Attempt,
+    TestCaseAttempt,
+    Badge,
+    Earned,
+)
+from django.http import JsonResponse
+from django.conf import settings
+
+time_zone = settings.TIME_ZONE
+
+logger = logging.getLogger(__name__)
+del logging
+
+LOGGING = {
+    'version': 1,
+    'disable_existing_loggers': False,
+    'incremental': True,
+    'root': {
+        'level': 'DEBUG',
+    },
+}
+
+
+def add_points(question, profile, attempt):
+    """add appropriate number of points (if any) to user's account after a question is answered"""
+    num_attempts = Attempt.objects.filter(question=question, profile=profile)
+    is_first_correct = len(Attempt.objects.filter(question=question, profile=profile, passed_tests=True)) == 1
+    logger.warning(num_attempts)
+    logger.warning(is_first_correct)
+    points_to_add = 0
+
+    # check if first passed
+    if attempt.passed_tests and is_first_correct:
```

Comment:
`is_first_correct`: Is this to prevent people repeatedly submitting the same solution over and over for points? Might need to explain that with a few more words